app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -163,557 +222,642 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+
+                List<SessionToken> sessionTokenData = activity.sessionData.getSessionTokenData();
+                signatureString = new ArrayList<>();
+                for (SessionToken item : sessionTokenData) {
+                    if (item != null) {
+                        LinkedHashMap<String, Object> msg = new LinkedHashMap<>();
+                        msg.put("data", item.getToken());
+                        msg.put("sig", item.getSignature());
+                        String jsonData = gson.toJson(msg);
+                        signatureString.add(jsonData);
                     }
                 }
-            });
-        });
 
-        binding.cleanupRequests.setOnClickListener(view1 -> {
-            try {
-                SharetransferModule.cleanupRequest(activity.tKey);
-                requireActivity().runOnUiThread(() -> {
-                    binding.requestNewShare.setEnabled(true);
-                    binding.cleanupRequests.setEnabled(false);
-                    binding.lookForRequests.setEnabled(false);
-                    binding.requestStatusCheck.setEnabled(false);
-                    activity.transferKey = null;
-                    activity.transferStorage = null;
-                    activity.transferProvider = null;
-                    Snackbar snackbar = Snackbar.make(view1, "Successfully cleaned up the requests", Snackbar.LENGTH_LONG);
-                    snackbar.show();
-                });
-            } catch (RuntimeError e) {
-                renderError(e);
-            }
-        });
+                // node details
+                FetchNodeDetails nodeManager = new FetchNodeDetails(TorusNetwork.SAPPHIRE_DEVNET);
+                CompletableFuture<NodeDetails> nodeDetailResult = nodeManager.getNodeDetails(verifier, verifierId);
+                nodeDetail = nodeDetailResult.get();
+
+                // Torus Utils
+                TorusCtorOptions torusOptions = new TorusCtorOptions("Custom");
+                torusOptions.setNetwork(TorusNetwork.SAPPHIRE_DEVNET.toString());
+                torusOptions.setClientId("BG4pe3aBso5SjVbpotFQGnXVHgxhgOxnqnNBKyjfEJ3izFvIVWUaMIzoCrAfYag8O6t6a6AOvdLcS4JR2sQMjR4");
+                TorusUtils torusUtils = new TorusUtils(torusOptions);
+                String[] tssEndpoint = nodeDetail.getTorusNodeTSSEndpoints();
+                RssComm rss_comm = new RssComm();
 
-        binding.createThresholdKey.setOnClickListener(view1 -> {
-            showLoading();
-            try {
                 activity.tkeyStorage = new StorageLayer(false, "https://metadata.tor.us", 2);
-                activity.tkeyProvider = new ServiceProvider(false, activity.postboxKey);
-                activity.tKey = new ThresholdKey(null, null, activity.tkeyStorage, activity.tkeyProvider, null, null, false, false);
+                activity.tkeyProvider = new ServiceProvider(true, activity.postboxKey,true, verifier, verifierId, nodeDetail);
```

Comment:
Currently logging has no effect, can be left as false.
```suggestion
                activity.tkeyProvider = new ServiceProvider(false, activity.postboxKey,true, verifier, verifierId, nodeDetail);
```